### PR TITLE
Refactored aggregations

### PR DESCRIPTION
**Pull request by jfrommann**
Refactored aggregations: 
Now aggregations can (and must) be specified per search call. For this the interface *SearchAggregation* has been introduced (the prefix "Search..." was chosen due to a name clash with the Elasticsearch Aggregation class used in its SearchResponse class). The implementations are *TermsAggregation*, *DateHistogramAggregation*, *DateRangeAggregation* and *DefaultAggregation*. The latter can be useful to add aggregations without much knowledge about the details. Just an easy way to get some data, which fits in a lot of use cases.

Exposed Elasticsearch REST client via getRestClient() method in ElasticsearchService.

Added FindAllExpression, mainly for aggregations.

Refactored the FacetConverter a bit, so it could be useful when using the REST client directly.

RFC

### CODE REVIEW

File: src/main/java/de/picturesafe/search/parameter/SearchParameter.java [hunk 1]
```diff
@@ -197,8 +186,7 @@ public String toString() {
                 .append("ignoreState", ignoreState) //--
                 .append("ignoreDeleted", ignoreDeleted) //--
                 .append("language", language) //--
-                .append("aggregationFields", aggregationFields) //--
-                .append("defaultAggregationMaxCount", defaultAggregationMaxCount) //--
+                .append("aggregationFields", aggregations) //--
```
Comment (araithel) [inline src/main/java/de/picturesafe/search/parameter/SearchParameter.java:189]: "aggregationFields" -> "aggregations"

File: src/main/java/de/picturesafe/search/parameter/aggregation/DefaultAggregation.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.picturesafe.search.parameter.aggregation;
+
+import de.picturesafe.search.util.logging.CustomJsonToStringStyle;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+
+/**
```
Comment (araithel) [inline src/main/java/de/picturesafe/search/parameter/aggregation/DefaultAggregation.java:22]: Maybe better: ?

Aggregation with default parameters.
The type of the aggregation will be determined by the field type.

File: src/main/java/de/picturesafe/search/parameter/aggregation/TermsAggregation.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+     * @param order Order
+     * @return      The aggregation
+     */
+    public TermsAggregation order(Order order) {
+        this.order = order;
+        return this;
+    }
+
+    /**
+     * gets the order of the returned buckets.
```
Comment (araithel) [inline src/main/java/de/picturesafe/search/parameter/aggregation/TermsAggregation.java:80]: Minor: "gets" -> "Gets"

File: src/test/java/de/picturesafe/search/elasticsearch/connect/FacetIT.java [hunk 4]
```diff
@@ -100,15 +102,13 @@ public void tearDown() {
     @Test
     public void testTextFacet() {
         Expression expression = new FulltextExpression("wert");
-        QueryFacetDto queryFacetDto = new QueryFacetDto("caption", 10, 100);
-        List<QueryFacetDto> queryFacetDtos = Collections.singletonList(queryFacetDto);
+        List<TermsAggregation> queryFacetDtos = Collections.singletonList(TermsAggregation.field("caption"));
```
Comment (araithel) [inline src/test/java/de/picturesafe/search/elasticsearch/connect/FacetIT.java:105]: I would rename variable "queryFacetDtos" to "aggregations"

File: src/test/java/de/picturesafe/search/elasticsearch/connect/FacetIT.java [hunk 5]
```diff
@@ -118,9 +118,7 @@ public void testTextFacet() {
     @Test
     public void testFacetResolverOnTextFacet() {
         final Expression expression = OperationExpression.and(new FulltextExpression("wert"), new ValueExpression("caption", "caption1"));
-        final QueryFacetDto queryFacetDto = new QueryFacetDto("facetResolved", 10, 100);
-        final List<QueryFacetDto> queryFacetDtos = new ArrayList<>();
-        queryFacetDtos.add(queryFacetDto);
+        final List<TermsAggregation> queryFacetDtos = Collections.singletonList(TermsAggregation.field("facetResolved"));
```
Comment (araithel) [inline src/test/java/de/picturesafe/search/elasticsearch/connect/FacetIT.java:121]: I would rename variable "queryFacetDtos" to "aggregations"

File: src/main/java/de/picturesafe/search/parameter/aggregation/DefaultAggregation.java (hunk 2 above)
Comment (jfrommann) [reply src/main/java/de/picturesafe/search/parameter/aggregation/DefaultAggregation.java:22]: OK

**Pull request merged**